microk8s-resources/wrappers/microk8s-reset.wrapper

```diff
@@ -2,31 +2,30 @@
 
 set -eu
 
+KUBECTL=/snap/bin/microk8s.kubectl
+
 clean_cluster() {
   echo "Calling clean_cluster"
   # Clean the cluster
-  NSES=$(/snap/bin/microk8s.kubectl get ns -o=custom-columns="NAME:metadata.name" --no-headers 2>&1 | tr '\n' ' ')
+  NSES=$($KUBECTL get ns -o=name | awk -F/ '{print $2}' | paste -s --delimiters=' ')
   for NS in $NSES
   do
-    echo "Cleaning service, deployments and pods in namespace $NS"
-    # delete all replicationcontrollers
-    /snap/bin/microk8s.kubectl delete --all replicationcontrollers --namespace="$NS" 2>&1 > /dev/null
-    # delete all daemonsets
-    /snap/bin/microk8s.kubectl delete --all daemonsets --namespace="$NS" 2>&1 > /dev/null
-    # delete all services
-    /snap/bin/microk8s.kubectl delete --all services --namespace="$NS" 2>&1 > /dev/null
-    # delete all deployments
-    /snap/bin/microk8s.kubectl delete --all deployments --namespace="$NS" 2>&1 > /dev/null
-    # delete all pods
-    /snap/bin/microk8s.kubectl delete --all pods --namespace="$NS" 2>&1 > /dev/null
+    echo "Cleaning resources in namespace $NS"
+    # Remove all resource types except the standard k8s apiservices themselves
+    $KUBECTL delete --all $($KUBECTL api-resources -o name --verbs=delete --namespaced=true | grep -v '^apiservices$' | paste -s --delimiters=',') --namespace="$NS"
   done
+  REMOVEABLE_NSES=$($KUBECTL get ns -o=name | awk -F/ '{if ($2 !~ /^kube-|^default$/) print}' | paste -s --delimiters=' ')
```

Comment:
Why do you say that the non-removable are all the namespaces starting with `kube-`? I think only `kube-system` and `kube-public` are the ones you cannot delete, but I could not find anything in the docs.

Reply:
May have over-generalized, I can make the non-removable namespaces specifically `kube-system` and `kube-public`.

Reply:
That would have been great, thank you.

Other than that the PR LGTM. Testing went fine. We should be able to merge this patch right after a few pending PRs.

Reply:
Updated, PTAL, thanks!